vpr/src/power/power.cpp

```diff
@@ -825,7 +826,7 @@ static void power_usage_routing(t_power_usage* power_usage,
                     auto next_node = device_ctx.rr_nodes[node.edge_sink_node(edge_idx)];
                     t_rr_node_power* next_node_power = &rr_node_power[node.edge_sink_node(edge_idx)];
 
-                    switch (next_node.type()) {
+                    switch (rr_graph.node_type(RRNodeId(node.edge_sink_node(edge_idx)))) {
```

Comment:
the duplication of `node.edge_sink_node(edge_idx)` and above when we get the next_node is a bit concerning as it might get out of sync and it is harder to see that we're talking about the same node.
We should probably have a `const next_node_id = node.edge_sink_node(edge_idx);` and use that when getting next_node and in this switch.

(there might be other places in this change where this might be a good idea)

Reply:
I have updated this instance. Let me know if you had something else in mind than what I have changed it to.